src/compiler/compilercontroler.cpp

```diff
@@ -0,0 +1,156 @@
+#include <QDirIterator>
+#include "filemanager.h"
+#include <QPlainTextEdit>
+#include <consolewindow/consolewindow.h>
+#include <QString>
+#include <QThread>
+#include <QProcess>
+#include <QSettings>
+
+CompilerControler::CompilerControler()
+{
+    consoleProvider = new ConsoleServiceProvider;
+}
+
+QString CompilerControler::compilerPath() const
+{
+    return mCompilerPath;
+}
+
+void CompilerControler::setCompilerPath(QString compilerPath)
+{
+    mCompilerPath = compilerPath;
+}
+
+ConsoleServiceProvider *CompilerControler::getConsoleProvider() const
+{
+    return consoleProvider;
+}
+
+void CompilerControler::setConsoleProvider(ConsoleServiceProvider *value)
+{
+    consoleProvider = value;
+}
+
+QString CompilerControler::getCurrentCompilerPath() const
+{
+    QString rCurrentCompiler = QString();
+    QSettings settings;
+    if (settings.contains(compilersListNamesInSettings))
+    {
+        QStringList currentCompilers = settings.value(compilersListNamesInSettings).toStringList();
+        if (!currentCompilers.empty())
+        {
+            QString firstCompilerPath = *currentCompilers.begin();
+            rCurrentCompiler = firstCompilerPath;
+        }
+    }
+    return rCurrentCompiler;
+}
+
+QString CompilerControler::getProjectPath() const
+{
+    return mProjectPath;
+}
+
+void CompilerControler::setProjectPath(const QString &projectPath)
+{
+    mProjectPath = projectPath;
+}
+
+void CompilerControler::runCompilation()
+{
+
+    auto currentCompilerName = getCurrentCompilerPath();
+    sourceFilesPathes.clear();
+    removeAllExecutableAndObjectsFiles();
+
+    auto buildDirectoryPath = mProjectPath + "/" + buildFolderName;
+    QDir buildDirectory(buildDirectoryPath);
+    if (!buildDirectory.exists())
+    {
+        buildDirectory.mkdir(buildDirectoryPath);
+    }
+
+    auto makeFilePath = buildDirectoryPath + "/" + makeFileName;
+
+    getAllSourceFilesFromTheProjectDirectory();
+
+    FileManager fileManager;
+    fileManager.createFile(makeFilePath);
+    fileManager.writeToFile(makeFilePath, createMakeFileContent(getExecutibleFileName()));
+
+    consoleProvider->setWorkingDirectory(buildDirectoryPath);
+    consoleProvider->runConsoleCommand(currentCompilerName.append(" -f MakeFile"));
+}
+
+QString CompilerControler::getExecutibleFileName() const
+{
+    auto fullProjectPathParts = mProjectPath.split('/');
+    return fullProjectPathParts.last();
+}
+
+void CompilerControler::removeAllExecutableAndObjectsFiles()
+{
+    // we should remove all files which will be created during the next compilation
+    QDir dir(mProjectPath);
+    dir.setNameFilters(QStringList() << "*" + objectiveFileExtension
+                       << "*" + executableFileExtension
+                       << makeFileName);
+    dir.setFilter(QDir::Files);
+    foreach (QString dirFile, dir.entryList())
+    {
+        dir.remove(dirFile);
+    }
+}
+
+QString CompilerControler::createMakeFileContent(const QString &executibleFileName) const
```

Comment:
Think about case, when user' directory is: **"C:\\\\Users\\\\faitc\\\\OneDrive - lpnu.ua\\\\Cplusplus STUFF"**
Other words, has **-** sign inside ;)